Wikipedia/Code/WKWebView+WMFWebViewControllerJavascript.m

```diff
@@ -18,7 +19,11 @@ - (void)wmf_collapseTablesForArticle:(MWKArticle *)article {
 
 - (void)wmf_addEditPencilsForArticle:(MWKArticle *)article {
     if (!article.isMain) {
-        [self evaluateJavaScript:@"window.wmf.editButtons.add(document);" completionHandler:nil];
+        [self evaluateJavaScript: [NSString stringWithFormat:@""
+                                   "window.wmf.editButtons.add(document);"
+                                   "document.querySelectorAll('.pagelib_edit_section_link').forEach(anchor => {anchor.href = '%@'});"
```

Comment:
Had to set the edit pencil href to an identifiable string because:
- the transform doesn't set an href for the edit pencil anchors
- when the browser sees the href is unset it seems to assume the url of the current article (hence peeking the pencil showing the same article)
- WKPreviewElementInfo only provides access to `linkURL`, so there's no other thing you can check in `- (BOOL)webView:(WKWebView *)webView shouldPreviewElement:(WKPreviewElementInfo *)elementInfo`